src/api/routes/transcripts.ts

```diff
@@ -52,8 +54,72 @@ const list = procedure
   return g.transcripts;
 });
 
+const getNameMap = procedure
+  .use(authUser())
+  .input(z.string())
+  .query(async ({ input: transcriptId }) => {
+
+    const summaries = await db.Summary.findAll({ where: { transcriptId } });
+
+    // find all handlebar from summaries under one transcript
+    let handlebars: string[] = [];
+    for (let s of summaries) {
+      const matches = s.summary.match(/{{(.*?)}}/g);
+      // parse and find handlebar values but get rid of the repeating values
+      if (matches) {
+        handlebars = [...handlebars, ...matches.map(match => match.slice(2, -2).trim())]
+        .reduce((acc, cur) => acc.includes(cur) ? acc : [...acc, cur], handlebars);
+      }
+    }
+
+    let nameMap: Record<string, string> = {};
+    // create a list of namemap of itself, otherwise when handlebars compile it will return empty
+    nameMap = handlebars.reduce((o, key) => ({ ...o, [key]: key }), {});
+
+    const snm = await db.SummaryNameMapping.findAll({
+      where: { handlebarName: handlebars },
+      include: [{
+        model: db.User,
+        attributes: ['name']
+      }],
+    });
+
+    if (snm) {
+      for (const nm of snm) {
+        invariant(nm.user.name);
+        nameMap[nm.handlebarName] = nm.user.name;
+      }
+    }
+
+    return nameMap;
+
+  });
+
+const updateNameMap = procedure
+  .use(authUser())
+  .input(z.object({}).catchall(z.string()))
```

Comment:
don't do this. use explicit type definition like `z.record(z.string(), ...)` instead